src/main/java/org/mqttbee/api/mqtt/mqtt5/message/publish/Mqtt5WillPublishBuilder.java

```diff
@@ -22,29 +22,37 @@
 import org.mqttbee.annotations.Nullable;
 import org.mqttbee.api.mqtt.datatypes.MqttQoS;
 import org.mqttbee.api.mqtt.datatypes.MqttTopic;
+import org.mqttbee.api.mqtt.datatypes.MqttTopicBuilder;
 import org.mqttbee.api.mqtt.datatypes.MqttUTF8String;
 import org.mqttbee.api.mqtt.mqtt5.datatypes.Mqtt5UserProperties;
+import org.mqttbee.api.mqtt.mqtt5.datatypes.Mqtt5UserPropertiesBuilder;
 import org.mqttbee.mqtt.message.publish.MqttWillPublish;
 import org.mqttbee.mqtt.util.MqttBuilderUtil;
 import org.mqttbee.util.MustNotBeImplementedUtil;
 import org.mqttbee.util.UnsignedDataTypes;
 
 import java.nio.ByteBuffer;
+import java.util.function.Function;
 
 import static org.mqttbee.mqtt.message.publish.MqttWillPublish.DEFAULT_DELAY_INTERVAL;
 
 /**
  * @author Silvio Giebl
  */
-public class Mqtt5WillPublishBuilder extends Mqtt5PublishBuilder {
+public class Mqtt5WillPublishBuilder<P> extends Mqtt5PublishBuilder<P> {
+
+    private final Function<? super Mqtt5WillPublish, P> parentConsumer;
```

Comment:
why is this needed?

Reply:
Because MqttWillPublishBuilder extends MqttPublishBuilder